src/Microsoft.AspNetCore.SystemWebAdapters/Build/NuGetContentFiles.targets

```
@@ -0,0 +1,53 @@
+<Project>
+
+  <!--
+      Add NuGet content to the build output if specified:
+      <PackageReference Include="..." CopyContent="true" ... />
+       -->
+  <Target Name="AddNugetContentFiles" BeforeTargets="AssignTargetPaths">
+
+    <!-- Identity PackageReference and paths to packages -->
+    <ItemGroup>
+      <_PackageReferenceProperties Include="Pkg$([System.String]::Copy('%(PackageReference.Identity)').Replace('.','_'))">
+        <PackageName>%(PackageReference.Identity)</PackageName>
+        <CopyContent>%(PackageReference.CopyContent)</CopyContent>
+      </_PackageReferenceProperties>
+      <_PackageReferenceProperties Update="@(_PackageReferenceProperties)" PackageRootPath="$(%(_PackageReferenceProperties.Identity))" />
+      <_PackageReferenceCopyContent Include="@(_PackageReferenceProperties)" Condition="'%(CopyContent)' == 'true'" />
```

If users will always be using a new enough version of MSBuild you can use intrinsic item functions: https://learn.microsoft.com/en-us/visualstudio/msbuild/item-functions?view=vs-2022#intrinsic-item-functions
```suggestion
      <_PackageReferenceCopyContent Include="@(_PackageReferenceProperties->WithMetadataValue('CopyContent', 'true'))" />
```

Functions like `WithMetadataValue()` were added in recent version of 17.x though, so if a user has 15.x it would fail.  Just thought I'd mention it in case you can use it.

---

I don't know if we could make that assumption. Most likely, this only would be used for ASP.NET Core apps, but we currently support ASP.NET 6.

---

I like that syntax better though - seems cleaner :)